src/link/Elf.zig

```diff
@@ -1631,7 +1631,22 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
         if (self.base.options.each_lib_rpath) {
             var test_path = std.ArrayList(u8).init(self.base.allocator);
             defer test_path.deinit();
+
+            const cross_target = std.zig.CrossTarget.fromTarget(target);
+            const target_info = try std.zig.system.NativeTargetInfo.detect(self.base.allocator, cross_target);
```

Comment:
Using `std.zig.system.NativeTargetInfo.detect` or related does not belong in the linker code and should only ever be invoked in `main.zig`. Please have a look at how we handle detection of macOS SDK for example (see [this line](https://github.com/ziglang/zig/blob/04f3d9301798e41ac8272822328914073e01a6ab/src/main.zig#L2064)). If you want to filter the `native_paths` based on the detected native paths, this should be done in `main.zig` when passing the info to `Compilation.zig`. Lemme know if that makes sense, or if you need any guidance or clarification.

Reply:
Yeah, apologies for that.

At the time I couldn't find the starting point I was looking for and took this as the best location to get something off the ground.

I've been delayed a bit from working on this again, but I've now looked through the code and figured its better to remove what was the cause of this instead of adding new code